EIPS/eip-6672.md

```diff
@@ -0,0 +1,247 @@
+
+- Operator wallet address, i.e. `_operator`
+
+    It's possible that there are more than one party who would like to use the same NFT for redemption. For example, MisterPunks NFTs are eligible to be redemeed for both Event-X and Event-Y tickets, and each event's ticket redemption is handled by a different operator.
+
+- Token identifier, i.e. `_tokenId`
+
+    Each NFT holder will have different redemption records created by the same operator. Therefore, it's important to use token identifier as one of the keys.
+
+- Redemption identifier, i.e. `_redemptionId`
+
+    Using `_redemptionId` as one of the keys enables NFT holders to redeem the same NFT to the same operator in multiple campaigns. For example, Operator-X has 2 campaigns, i.e. campaign A and campaign B, and both campaigns allow for MisterPunks NFTs to be redemeed for physical action figures. Holder of MisterPunk #7 is eligible for redemption in both campaigns and each redemption is recorded with the same `_operator` and `_tokenId`, but with different `_redemptionId`.
+
+### Metadata Choices (metadata extension)
+
+To provide additional information about a redemption, `redemptions` key-value pairs is added to the "[ERC-721](./eip-721.md) Metadata Extension". The key format for the redemptions key-value pairs MUST be standardized as `operator-tokenId-redemptionId`, where `operator` is the operator wallet address, `tokenId` is  the identifier of the token that has been redeemed, and `redemptionId` is the redemption identifier. The same combination of keys is used for the redemption flag, and the same rationale applies.
+
+The value of the key `operator-tokenId-redemptionId` is an object that contains the `status` and `description` of the redemption.
+
+- Redemption status, i.e. `status`
+
+    The redemption status can have a more granular level, rather than just being a flag with a `true` or `false` value. For instance, in cases of physical goods redemption, we may require the redemption status to be either `redeemed`, `paid`, or `shipping`. It is RECOMMENDED to use a string enum that is comprehensible by both the operator and the marketplace or any other parties that want to exhibit the status of the redemption.
+
+- Description of the redemption, i.e. `description`
+
+    The `description` can be used to provide more details about the redemption, such as information about the concert ticket, a detailed description of the action figures, and more.
+
+## Backwards Compatibility
+
+This standard is compatible with [ERC-721](./eip-721.md).
+
+## Reference Implementation
+
+Here's an example of an [ERC-721](./eip-721.md) that includes the Multi-Redeemable ([IERC-6672](./eip-6672.md)) extension:
+
+```solidity
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.16;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "./interfaces/IERC6672.sol";
+
+contract ERC6672 is ERC721, IERC6672 {
```

Comment:
```suggestion
contract ERC6672 is ERC721, ERC6672 {
```

Reply:
I believe this is correct since ERC6672 is the implementation of the interface IERC6672

Reply:
The interface name (as defined [here](https://github.com/ethereum/EIPs/pull/6672/files#diff-09ca8486e6261457e9870d748205103913ecfceabbca2c30c117c6c172f2ce03R41)) is `ERC6672`. Should probably keep that consistent in your reference implementation.

Reply:
Aha nice catch, alrighty I have changed the interface contract code in the specification to `IERC`.